streams/src/main/java/org/apache/kafka/streams/state/internals/StoreQueryUtils.java

```diff
@@ -335,6 +342,42 @@ private static <R> QueryResult<R> runWindowRangeQuery(final Query<R> query,
         }
     }
 
+    @SuppressWarnings("unchecked")
+    private static <R> QueryResult<R> runVersionedKeyQuery(final Query<R> query,
+                                                           final PositionBound positionBound,
+                                                           final QueryConfig config,
+                                                           final StateStore store) {
+        if (store instanceof VersionedKeyValueStore) {
```

Comment:
Do we need this check? Seems our code ensures that we call this method only is this condition is true. -- If we call it incorrectly a hard crash would be preferred to clearly surface the bug, and not masked it, and thus make it harder to debug.

Reply:
> Do we need this check? Seems our code ensures that we call this method only is this condition is true. -- If we call it incorrectly a hard crash would be preferred to clearly surface the bug, and not masked it, and thus make it harder to debug.

I think yes. When we are calling `StateQueryRequest.inStore("my_store_name").withQuery(myVersionedKeyQuery)`, how does it know that the `my_store_name` is of which type? Ok, if we remove this condition, then there will be a hard crash but there is no suitable error message to instruct the user on how to debug it or where he should start debugging and looking into it.

Reply:
Ah. you are right. We need this here. My bad.